types/jasminewd2/index.d.ts

```diff
@@ -5,7 +5,7 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
-/// <reference types="jasmine" />
+/// <reference types="jasmine/v2" />
```

Comment:
With this change published, I have errors similar to `Argument of type 'Whatever' is not assignable to parameter of type 'string'.` in the parameter of `withArgs()`. 
I have yet to come up with a bare reproduction, however any tips are welcome in the meantime.

Reply:
I'm very sorry for the mess created around - it's always annoying 😖 I had only good intentions to make jasmine usage better.

@gkalpak

> Wouldn't a major (or at least minor) version bumb be appropriate (based on this doc)?

If you are speaking about `jasmine` library, then no. There are no breaking changes in the types itself and the previously compatible code is still compatible. Of course, new types are more stricter, but that's always the case when you improve type coverage.

The reason `jasminewd2` got affected is because that project was not configured correctly. They must have depended on `jasmine v2` instead of pretending being compatible with v3. See more detail [here](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/38656#issuecomment-535936695).

> According to the docs, jasminewd2/tsconfig.json should have been updated instead, mapping jasmine to jasmine/v2.

That's for spotting it - I read that article, but missed the part about dependent packages. I'll try to give it a go. Anyway, I think it still could have limitations according to [this answer](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/38656#issuecomment-536116451).

@AndrejEPAM

> With this change published, I have errors similar to Argument of type 'Whatever' is not assignable to parameter of type 'string'. in the parameter of withArgs().

Indeed now we check types for the parameters. You should pass only compatible types. In your case typescript is not making widening, which has been fixed in later versions of TS. Notice, if you don't care about value, just use `jasmine.anything()` matcher.

> Also i'm no longer able to use jasmine.ObjectContaining() to match a partial representation of a type in withArgs().

Strange, as we have test case for that. I've just tried again and the code below worked as a charm:

```ts
const spy = jasmine.createSpyObj<{
    method: (arg: { prop1: string, prop2: number }) => void }
    >(['method']);

expect(spy.method)
    .toHaveBeenCalledWith(jasmine.objectContaining({ prop1: "forty-two" }));
```

If everything is fine with code and it fails, please create a standalone GitHub issue and share scenarios, so they could be investigated in more detail.

Notice, you could also use `NonTypedSpyObj<T>` instead of `SpyObj<T>` if you would like to opt-out from type checking. Sometimes it's required if method signature is tricky and typescript cannot deduce type correctly (e.g. you have multiple overloads).

Reply:
@zvirja thanks for the valuable tips, will try them and create reproduction, if still applicable. I was using `jasmine.objectContaining` within `withArgs`, specifically, it worked like a charm, but it may be unintended/unsupported use of `objectContaining`.

Reply:
@AndrejEPAM Oooh, you mean `withArgs`. Silly me, overlooked that twice, sorry 😖 See it now. Yes, that's a regression and it should be fixed - I overlooked that you could specify argument matchers there 😟

Reply:
@AndrejEPAM JFYI Created a PR to allow arg matchers in `withArgs()` method: #38738.

Reply:
I have a similar issue with 3.4.2. This code no longer compiles:

```ts
expect(ga).toHaveBeenCalledWith("send", jasmine.objectContaining({
    hitType: "event",
    eventCategory: category,
    eventAction: action,
    eventLabel: label
}));
```